tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -101,21 +101,41 @@ int RelayHandler::open(const ACE_INET_Addr& local)
 int RelayHandler::handle_input(ACE_HANDLE)
 {
   ACE_INET_Addr remote;
-  iovec iov;
-  const auto bytes = socket_.recv(&iov, remote);
+  int inlen = 65536; // Default to maximum datagram size.
+  const int iov_size = 1;
+  iovec iov[iov_size];
+  OpenDDS::DCPS::Message_Block_Shared_Ptr buffer;
+
+#ifdef FIONREAD
+  if (ACE_OS::ioctl (get_handle(),
```

Comment:
```suggestion
  if (ACE_OS::ioctl(handle,
```
Given that `handle` is the parameter to this method.